From c612c6cd48130c6385294f684222d2bc19b53f50 Mon Sep 17 00:00:00 2001
From: yaozenhu <yaozenhu@realtek.com>
Date: Mon, 16 Jan 2017 07:28:25 +0000
Subject: [PATCH 7/7] [fix bug] DHCKYLIN-1884 [root cause] Vout end state after
 Vdec state [solution] let Vout end state before Vdec state [reviewer] Paul

Change-Id: I410c954a91de0f54bb5532f4b5b0194386db1e4e
---
 libs/gst/base/gstbasesink.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libs/gst/base/gstbasesink.c b/libs/gst/base/gstbasesink.c
index afcbf88..97e7007 100644
--- a/libs/gst/base/gstbasesink.c
+++ b/libs/gst/base/gstbasesink.c
@@ -5237,8 +5237,19 @@ gst_base_sink_change_state (GstElement * element, GstStateChange transition)
         GST_DEBUG_OBJECT (basesink, "PAUSED to READY, don't need_preroll");
       }
       GST_BASE_SINK_PREROLL_UNLOCK (basesink);
+
+      if (bclass->stop) {
+        if (!bclass->stop (basesink)) {
+          GST_WARNING_OBJECT (basesink, "failed to stop");
+        }
+      }
+      gst_base_sink_set_last_buffer (basesink, NULL);
+      gst_base_sink_set_last_buffer_list (basesink, NULL);
+      priv->call_preroll = FALSE;
+
       break;
     case GST_STATE_CHANGE_READY_TO_NULL:
+/*
       if (bclass->stop) {
         if (!bclass->stop (basesink)) {
           GST_WARNING_OBJECT (basesink, "failed to stop");
@@ -5247,6 +5258,7 @@ gst_base_sink_change_state (GstElement * element, GstStateChange transition)
       gst_base_sink_set_last_buffer (basesink, NULL);
       gst_base_sink_set_last_buffer_list (basesink, NULL);
       priv->call_preroll = FALSE;
+*/
       break;
     default:
       break;
-- 
2.7.4

